### PR TITLE
Add dotnet nuget why documentation

### PR DESCRIPTION
**Pull request by pragnya17**
## Summary

Add documentation for `dotnet nuget why`. The implementation of this command is still in progress: https://github.com/NuGet/Home/issues/11782.

### CODE REVIEW

File: docs/core/tools/dotnet-nuget-why.md
Comment (tdykstra) [inline docs/core/tools/dotnet-nuget-why.md:3]:
```suggestion
description: The 'dotnet nuget why' command shows the dependency graph of a package.
```

File: docs/core/tools/dotnet-nuget-why.md
Comment (tdykstra) [inline docs/core/tools/dotnet-nuget-why.md:64]: The --outdated option isn't in the doc here.

File: docs/core/tools/dotnet-nuget-why.md
Comment (tdykstra) [inline docs/core/tools/dotnet-nuget-why.md:24]: This would be easier to understand with actual command output using real package/project names rather than packageA and B and projectA and B.

File: docs/core/tools/dotnet-nuget-why.md
Comment (tdykstra) [inline docs/core/tools/dotnet-nuget-why.md:8]: Which preview is this in? We will want to make sure it doesn't get merged before it's available, and this applies-to text should be updated to .NET 7 Preview SDK.

File: docs/core/tools/dotnet-nuget-why.md
Comment (tdykstra) [inline docs/core/tools/dotnet-nuget-why.md:74]:
```suggestion
- Print dependency graph of `packageA`. The package has dependencies in multiple projects so all dependencies are printed by default:
```

But the command only looks at one project per the arguments section, so it's not clear what this means by referring to multiple projects.

File: docs/core/tools/dotnet-nuget-why.md
Comment (pkulikov) [inline docs/core/tools/dotnet-nuget-why.md:24]:
```suggestion
The `dotnet nuget why` command provides a way to print out the dependency graph for a package to allow users to understand the nature of top-level packages and their transitive dependencies. This command is dependent on the `project.assets.json` file being present in the project. The following example shows the output of the `dotnet nuget why` command for `packageA` which has dependencies in `projectNameA` and `projectNameB`:
```

**Review by tdykstra — CHANGES_REQUESTED** [review]
Waiting on responses to earlier review comments.

**Comment by BillWagner** [comment]
Hi @pragnya17 

Can you address the comments on this PR? Once you do that, we can merge it.

**Comment by tdykstra** [comment]
Closing due to no response from PR author.

**Pull request closed without merging**
